mm2src/adex_cli/src/adex_config.rs

```diff
@@ -151,17 +151,30 @@ impl AdexConfigImpl {
     }
 
     fn write_to(&self, cfg_path: &Path) -> Result<()> {
-        let adex_path_str = cfg_path
+        let komodefi_path_str = cfg_path
             .to_str()
             .ok_or_else(|| error_anyhow!("Failed to get cfg_path as str"))?;
-        rewrite_json_file(self, adex_path_str)?;
+        rewrite_json_file(self, komodefi_path_str)?;
         #[cfg(unix)]
         {
-            set_file_permissions(adex_path_str, CFG_FILE_PERM_MODE)?;
+            Self::warn_on_insecure_mode(komodefi_path_str)?;
         }
         Ok(())
     }
 
+    #[cfg(unix)]
+    fn warn_on_insecure_mode(file_path: &str) -> Result<()> {
+        let perms = fs::metadata(file_path)?.permissions();
+        let mode = perms.mode() & 0o777;
```

Comment:
Did you intentionally use the bitwise operator here? Why `perms.mode() != CFG_FILE_PERM_MODE` is not enough?

Reply:
I did it like that to get the first byte of mode out of the focus the File Type